src/app/rpmostree-builtin-status.cxx

```diff
@@ -864,10 +861,15 @@ print_one_deployment (RPMOSTreeSysroot *sysroot_proxy,
               g_string_append (str, pkgname);
             }
           g_string_append_printf (str, " %s", evr);
+
+          g_ptr_array_add (active_removals_grouped, g_strdup (str->str));
+          g_string_erase (str, 0, -1);
```

Comment:
This call deletes all the data, right?  So I'm a bit confused...why are we keeping around `str` here at all?

Reply:
Consistency with the current code base. Implementations around this allocate a buffer outside of the loop, then in the loop fill, use the buffer and erase it for the next loop. Proposal: Move autoptr of the buffer into the loop scope to make it's intention clear and avoid misuse from outer scope + rename `str` to `buf` (seen this in other places of the codebase). What do you think?

Reply:
I'll try to clean up the string buffer handling a little bit with my next commit - the handling for ReplacedBasePackages is not "trivial". One has to see that the positive if branch ends with a truncate to size 0. That's why the buffer can be reused without an erase in between the two loops.